modules/engines/engine-xsjob/src/main/java/com/sap/xsk/xsjob/ds/transformer/XSKCronToQuartzCronTransformer.java

```diff
@@ -88,4 +88,17 @@ private String parseDayOfWeekElement(String dayOfWeekElement) throws ParseExcept
 
     throw new XSKCronExpressionException(String.join(" ", xskCronExpressionArr), XSK_CRON_DAY_OF_WEEK);
   }
+
+  private String checkDayOfWeekAndDayOfMonth(){
+    final String[] xskCronExpressionDayOfWeekArr = {"MON", "TUE", "WED", "THU", "FRI", "SAT", "SUN", "1", "2", "3", "4", "5", "6", "7"};
+
+    if (xskCronExpressionArr[XSK_CRON_DAY].equalsIgnoreCase("*")){
+      for (String dayOfWeek : xskCronExpressionDayOfWeekArr ){
+        if (xskCronExpressionArr[XSK_CRON_DAY_OF_WEEK].contains(dayOfWeek)){
+          return "?";
+        }
+      }
+    }
```

Comment:
this can be simplified I think as it's difficult to read

1. `xskCronExpressionArr[XSK_CRON_DAY].equalsIgnoreCase("*")` can be assigned to `isEveryDay`
2. the for loop which iterates can be translated to stream with `anyMatch` where the predicate checks for `containsIgnoreCase` --> the result can be assigned to variable `hasDayOfWeek`

3. then if `isEveryDay && hasDayOfweek`  --> return